internal/builder/writer/human_readable.go

```diff
@@ -283,29 +283,38 @@ func buildpacksOutput(buildpacks []dist.BuildpackInfo, builderName string) (stri
 		return fmt.Sprintf("%s  %s\n", output, none), warnings, nil
 	}
 
-	tabWriterBuf := bytes.Buffer{}
-	spaceStrippingWriter := &trailingSpaceStrippingWriter{
-		output: &tabWriterBuf,
-	}
+	var (
+		tabWriterBuf         = bytes.Buffer{}
+		spaceStrippingWriter = &trailingSpaceStrippingWriter{
+			output: &tabWriterBuf,
+		}
+		buildpacksTabWriter = tabwriter.NewWriter(spaceStrippingWriter, writerMinWidth, writerPadChar, buildpacksTabWidth, writerPadChar, writerFlags)
+		dashOrValue         = func(str string) string {
+			if str == "" {
+				return "-"
+			}
+			return str
+		}
```

Comment:
nit: This function, though tiny, ends up getting used in a whole bunch of places. Do you think it's worth it to pull it out and share it between those? Or, at the very least, making it a standalone private func rather than a `var`

Reply:
I'm happy to be overruled here, but I did not think it was worth it.

I introduced it in 3 places, which is a smell, but it's so small that I couldn't justify a utility function. My rational is: if this were a more terse language, I would have done the same but use a tertiary operator.

Like I said, I'm happy to be overruled.

Reply:
> if this were a more terse language, I would have done the same but use a ternary operator.

😢 in `go`.

Reply:
I don't have a _strong_ opinion here but do think that this function looks a little 🥴. Given that go's design is geared towards making code comprehensible, I do think that this function although nice is counter towards clean comprehensible code.

The more idiomatic way would be a `strings` package with a default utility function:

```
strings.ValueOrDefault(b.Homepage, "-")
```

OR 

use long form:

```
homepage := b.Homepage
if homepage == "" {
   homepage = "-"
}
```

Reply:
I was swayed. I went for making a `strings` utility package. Hopefully it can be extended in the future.